docs/configuration/jobs.rst

```diff
@@ -173,6 +173,230 @@ Note that you will have to trigger replication manually using the ``zrepl signal
+``err_is_fatal=false`` logs the failed pre-edge invocation but does not affect subsequent hooks nor snapshotting itself.
+Post-edges are only invoked for hooks whose pre-edges ran without error.
+Note that hook failures for one filesystem never affect other filesystems.
+
+The optional ``timeout`` parameter specifies a period after which zrepl will kill the hook process and report an error.
+The default is 30 seconds and may be specified in any units understood by `time.ParseDuration <https://golang.org/pkg/time/#ParseDuration>`_.
+
+The optional ``filesystems`` filter which limits the filesystems the hook runs for. This uses the same |filter-spec| as jobs.
+
+``command`` Hooks
+~~~~~~~~~~~~~~~~~
+
+::
+
+
+    jobs:
+    - type: push
+      filesystems: {
+        "<": true,
+        "tmp": false
+      }
+      snapshotting:
+        type: periodic
+        prefix: zrepl_
+        interval: 10m
+        hooks:
+        - type: command
+          path: /path/to/hooks/zrepl-notify.sh
+          timeout: 30s
+          err_is_fatal: false
+        - type: command
+          path: /path/to/hooks/special-snapshot.sh
+          always_run: true
+          filesystems: {
+            "tank/special": true
+          }
+      ...
+
+
+The hook type ``command`` is the only currently supported hook type.
+Future versions of zrepl may support other hook types.
+The ``path`` to the hook executables must be absolute (e.g. ``/opt/bin/zrepl-notify.sh``).
+No arguments may be specified; create a wrapper script if zrepl must call an executable that requires arguments.
+Zrepl will call the hook both before and after the snapshot, but with different values of
+the ``ZREPL_HOOKTYPE`` environment variable; see below.
+
+Zrepl sets a number of environment variables for the hook processes:
+
+* ``ZREPL_HOOKTYPE``: either "pre_snapshot" or "post_snapshot"
+* ``ZREPL_FS``: the ZFS filesystem name being snapshotted
+* ``ZREPL_SNAPNAME``: the zrepl-generated snapshot name (e.g. ``zrepl_20380119_031407_000``)
+* ``ZREPL_DRYRUN``: set to ``"true"`` if a dry run is in progress so scripts can print, but not run, their commands
+
+The process standard output is logged at level INFO. Standard error is logged at level WARN.
+
+Note that the same hook may be invoked concurrently for multiple filesytems - namespace by filesystem 
+
+Examples:
+
+**Postgres Checkpoint Before Snapshot**
+
+.. code-block:: bash
+    :caption: postgresql-pre-snapshot.sh
+
+    #!/bin/sh -eu
+
+    ###
+    # Settings
+    ###
+    PSQL_TIMEOUT=$(($ZREPL_TIMEOUT - 1))
+
+    DRYRUN=""
+    [ "$ZREPL_DRYRUN" = "true" ] && DRYRUN="echo DRYRUN: "
+    [ "$ZREPL_HOOKTYPE" = "pre_snapshot" ] || exit
+
+    $DRYRUN /usr/local/bin/psql \
+      --echo-errors \
+      --no-password \
+      --no-psqlrc \
+      -U postgres \
+      -c "SET statement_timeout TO $PSQL_TIMEOUT; CHECKPOINT;"
+    # Exit with psql's exit code
+    exit $?
+
+**MySQL** ``LOCK_TABLES`` **During Snapshot**
```

Comment:
move those scripts to an example file, probably replace them with a template for people to copy instead

(we have built-in support for mysql and postgres, no need to show what these builtins do in such detail in the docs)